omnigan/transforms.py

```diff
@@ -81,6 +83,83 @@ def __call__(self, data):
         return {task: torch.flip(tensor, [3]) for task, tensor in data.items()}
 
 
+class RandomRotations:
+    def __init__(self, p=0.5, angle=5):
+        self.p = p
+        self.angle = angle
+
+    def cut_balck_edge(self, rotated, angle_selected):
+        if angle_selected < 0:
+            angle_selected = -angle_selected
+        angle_selected = angle_selected / 180 * pi
+        if angle_selected < 15:
+            tanAngle = angle_selected
+        else:
+            tanAngle = np.tan(angle_selected)
+        if len(rotated.shape) == 3:
+            rotated = rotated.unsqueeze(0)
+        _, _, h, w = rotated.shape
+        a = int(h * tanAngle / (tanAngle + 1))
+        c = int(w * tanAngle / (tanAngle + 1))
+        return rotated[:, :, c:-c, a:-a]
+
+    def mapping(self, rotated):
+        Dict = {
```

Comment:
where do the values come from ?

Reply:
By manual matching Lol

Reply:
but what do they correspond to ? Im just curious :)